### PR TITLE
Restructuring Module in the Upgrade guide

### PR DESCRIPTION
**Pull request by dubewarsagar**
Tuning {Project} with Predefined Profiles module needs Restructuring. 
The same has been done as suggested in the
https://bugzilla.redhat.com/show_bug.cgi?id=2124804#c1

https://bugzilla.redhat.com/show_bug.cgi?id=2124804

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3 -- PR #1642

### CODE REVIEW

**Comment by dubewarsagar** [comment]
@Lennonka could you please confirm if this restructuring is ok?

File: guides/doc-Upgrading_and_Updating/master.adoc [hunk 1]
```diff
@@ -96,6 +96,9 @@ endif::[]
 // Reclaiming Database Space
 include::common/modules/proc_reclaiming-postgresql-space.adoc[leveloffset=+3]
 
+// Updating Templates, Parameters, Lookup Keys and Values
+include::topics/post_upgrade_template_checks.adoc[leveloffset=+3]
+
```
Comment (Lennonka) [inline guides/doc-Upgrading_and_Updating/master.adoc:101]: We need to find out why this module was removed and therefore whether it's okay to put it back.

**Comment by Lennonka** [comment]
@ekohl We need to find the commit, in which it was removed to hopefully find out why it was done. Can you please assist Sagar to do that?

File: guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+The following error occurs because of old syntax remaining in files after the upgrade:
+
+----
+ undefined method '#params' for Host::Managed::Jail
+----
+
+.Fixing the outdated subscription_manager_registration snippet
+
+{Project} 6.4 onwards uses the `redhat_register` snippet instead of the `subscription_manager_registration` snippet.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc:29]: `6.4`?

**Comment by ekohl** [comment]
I ran `git log -S 'Updating Templates' -p` (`Updating Templates` is part of the title) which points to 23859bf7303b6822de5c2e1ddc7f6cf6bb49e1ad. Given what I wrote there I don't see why it should be added back.

File: guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
+[id="post_upgrade-updating-templates-parameters_{context}"]
+= Updating Templates, Parameters, Lookup Keys and Values
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc:2]: This was dropped in 23859bf7303b6822de5c2e1ddc7f6cf6bb49e1ad and should stay out.

File: guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+To ensure that this text replacement does not break or omit any variables in your files during the upgrade, check all templates, parameters, and lookup keys and values for the old syntax and replace manually.
+
+The following error occurs because of old syntax remaining in files after the upgrade:
+
+----
+ undefined method '#params' for Host::Managed::Jail
+----
+
+.Fixing the outdated subscription_manager_registration snippet
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/post_upgrade_template_checks.adoc:27]: This was dropped in 52b8ae194df55b5d9897278ca78a5450d1f5e53f and should also stay out

File: guides/doc-Upgrading_and_Updating/master.adoc [hunk 5]
```diff
@@ -96,6 +96,9 @@ endif::[]
...
 ifdef::katello,orcharhino,satellite[]
 //Tuning Satellite Server with Predefined Profiles
 include::common/modules/proc_tuning-with-predefined-profiles.adoc[leveloffset=+3]
```
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/master.adoc:104]: I think the Bugzilla was essentially that this should be:
```suggestion
include::common/modules/proc_tuning-with-predefined-profiles.adoc[leveloffset=+2]
```

I'm not sure what made you think the old bits needed to be reintroduced.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: guides/doc-Upgrading_and_Updating/master.adoc (hunk 5 above)
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/master.adoc:104]: We just needed to check. Thanks Ewoud.

Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/master.adoc:104]: No, it§s not desirable to change the leveloffset, because it's under the post-upgrade tasks.

**Comment by Lennonka** [comment]
@dubewarsagar In this case, no modification is needed elsewhere than 3.1, which is done in the other PR, and I'm closing this PR as unnecessary.

**Comment by dubewarsagar** [comment]
@Lennonka thank you for clarification on this.

**Pull request closed without merging**